bigtable/client/table.h

```diff
@@ -18,6 +18,8 @@
 #include "bigtable/client/data_client.h"
```

Comment:
I suspect a lot of these includes are redundant now.  We can clean that later though.